lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,225 @@
+        fall level. The percentile values are then interpolated to the
+        height of the point being considered. This constructs a
+        2-dimensional field of probabilities that snow will be falling at
+        each point in the orography field.
+    """
+
+    def __init__(self, reference_cube, output_name, inverse_ordering=False):
+        """
+        Initialise class.
+
+        Args:
+            reference_cube (iris.cube.Cube):
+                The percentiled field from which probabilities will be obtained
+                using the input cube. This cube should contain a percentiles
+                dimension, with fields of values that correspond to these
+                percentiles. The cube passed to the process method will contain
+                values of the same diagnostic (e.g. height) as this reference
+                cube.
+            output_name (str):
+                The name of the cube being created,
+                e.g.'probability_of_snowfall'.
+        Keyword Args:
+            inverse_ordering (bool):
+                Set True if the percentiled data increases in the opposite
+                sense to the percentile coordinate.
+                e.g.  0th Percentile - Value = 10
+                     10th Percentile - Value = 5
+                     20th Percenitle - Value = 0
+        """
+        self.reference_cube = reference_cube
+        self.output_name = output_name
+        self.inverse_ordering = inverse_ordering
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ProbabilitiesFromPercentiles2D: reference_cube: {}, '
+                  'output_name: {}, inverse_ordering: {}'.format(
+                      self.reference_cube, self.output_name,
+                      self.inverse_ordering))
+        return result
+
+    def create_probability_cube(self, cube):
+        """
+        Create a probability cube in which to store the calculated
+        probabilities.
+
+        Args:
+            cube (iris.cube.Cube):
+                The input cube to be used as a template.
+        Returns:
+            probability_cube (iris.cube.Cube):
+                The new cube with suitable metadata.
+        """
+        cube_format = next(cube.slices([cube.coord(axis='y'),
+                                        cube.coord(axis='x')]))
+        probabilities = cube_format.copy(data=np.full(cube_format.shape, 1.,
+                                                      dtype=float))
+        percentile_coordinate, = [coord.name() for coord in
+                                  cube_format.coords()
+                                  if 'percentile' in coord.name()]
+        if percentile_coordinate:
+            probabilities.remove_coord(percentile_coordinate)
+        probabilities.units = 1
+        probabilities.rename(self.output_name)
+        return probabilities
+
+    def percentile_interpolation(self, cube, reference_cube):
```

Comment:
I think the way that this is working is to find a pair of percentile values and a pair of height values that span the point, in this case a height point, that you want to interpolate to. I think that this is taking an alternative approach to https://github.com/metoppv/improver/blob/master/lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py#L406, so instead of bookending the height values with a value that is beyond the likely range,  the percentile is repeated (as done after catching the IndexError). Outliers are handled after the interpolation, see the usage of the above_top_band and below_bottom_band variables, instead of prior to the interpolation, as done in https://github.com/metoppv/improver/blob/master/lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py#L353. 

It would be good to decide if there is an optimum strategy this process, as perhaps naively, I would expect the percentile to probability operation to be a mirror image to the probability to percentile operation, and currently the operations are programmed quite differently.